context/app/static/js/components/entity-search/SearchWrapper/utils.js

```diff
@@ -0,0 +1,46 @@
+import { RefinementSelectFacet } from '@searchkit/sdk';
+
+function getDonorMetadataFilters(isDonor) {
+  const pathPrefix = isDonor ? '' : 'donor.';
+  const labelPrefix = isDonor ? '' : 'Donor ';
+
+  return [
+    {
+      field: `${pathPrefix}mapped_metadata.sex`,
+      label: `${labelPrefix}Sex`,
+    },
+    {
+      field: `${pathPrefix}mapped_metadata.race`,
+      label: `${labelPrefix}Race`,
+    },
+  ];
+}
+
+function createFacet({ field, label, ...rest }) {
```

Comment:
I think this is fine for now, but if it's helpful, here's some pseudo code for how this might be more OO:
```javascript
class FieldDef {
  static createFieldDef(id) {
    // Factory pattern
    const knownMetadataFields = ...
    if (knownMetadataFields.contains(id)) {
      return new MetadataFieldDef(id)
    } else {
      return new EntityFieldDef(id)
    }
  }
  constructor(id) {
    this.id = id;
  }
}

class MetadataFieldDef extends FieldDef {
  getType() {
    // pull from submodule
  }
  getDescription() {
    // pull from submodule
  }
}

class EntityFieldDef extends FieldDef {
  getType() {
    // hardcode here?
  }
  getDescription() {
    // hardcode here?
  }
}
```

Reply:
Thanks!